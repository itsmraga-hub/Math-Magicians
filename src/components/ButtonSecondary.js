import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import PropTypes from 'prop-types';

class ButtonSecondary extends React.PureComponent {
  render() {
    const { name, getButtonValue } = this.props;
    return <button type="button" onClick={getButtonValue} className="btn btn-secondary grid-item" data-id="" data-attribute="">{ name }</button>;
  }
}

ButtonSecondary.propTypes = {
  name: PropTypes.string.isRequired,
  getButtonValue: PropTypes.func.isRequired,
};

export default ButtonSecondary;
