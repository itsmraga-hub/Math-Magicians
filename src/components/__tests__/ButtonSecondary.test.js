import React from 'react';
import renderer from 'react-test-renderer';

import ButtonSecondary from '../ButtonSecondary';

const obj = {
  name: '=',
  getButtonValue: () => true,
};

it('<ButtonSecondary /> Renders properly', () => {
  const { name, getButtonValue } = obj;
  const tree = renderer.create(<ButtonSecondary
    name={name}
    getButtonValue={getButtonValue}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
