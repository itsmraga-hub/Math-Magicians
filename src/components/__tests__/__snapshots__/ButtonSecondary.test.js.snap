// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonSecondary /> Renders properly 1`] = `
<button
  className="btn btn-secondary grid-item"
  data-attribute=""
  data-id=""
  onClick={[Function]}
  type="button"
>
  =
</button>
`;
